datafusion/src/execution/dataframe_impl.rs

```diff
@@ -252,7 +252,12 @@ mod tests {
         let right = test_table()?.select_columns(&["c1", "c3"])?;
         let left_rows = left.collect().await?;
         let right_rows = right.collect().await?;
-        let join = left.join(right, JoinType::Inner, &["c1"], &["c1"])?;
+        let join = left.join(
+            right,
+            JoinType::Inner,
+            vec![Column::from_name("c1".to_string())],
```

Comment:
I think it might be useful to keep supporting the string syntax to keep the dataframe api simple, but also support qualified references.
Maybe the API could be used `column: Vec<Into<Column>>` and includes an implementation for `&str`?

Reply:
That's a great idea!

Reply:
Turns out it is not possible to use `Vec<impl Into<Column>>` in the signature because Dataframe is used as trait object. I changed the signature back to `&[&str]` for dataframe API and switched to `Vec<impl Into<Column>>` in the logical plan builder to keep the high level API simple.